imitation/util/buffer.py

```diff
@@ -0,0 +1,213 @@
+from typing import Optional, Tuple
+
+import numpy as np
+
+
+class Buffer:
```

Comment:
The type checking part of these annotations is less powerful than you might think. From my experience, `pytype` is completely in the dark whenever we touch the `numpy` module. It knows that `np.ndarray` is a type, but to my knowledge there are almost never instances where it can confirm whether a given variable fits `np.ndarray`, so it just let's everything pass.

The primary motivation for these annotations was to document instance variables.